notebooks/problem-sets/quantum_enigma_problem_set.ipynb

```diff
@@ -0,0 +1,381 @@
+    "problem_qc.x(1)\n",
+    "problem_qc.cx(2, 1)\n",
+    "problem_qc.x(0)\n",
+    "problem_qc.x(2)\n",
+    "problem_qc.cx(2, 0)\n",
+    "problem_qc.x(2)\n",
+    "\n",
+    "problem_qc.draw()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Try simplifying the circuit and rerun the calculation between each simplification to make sure you always get the same histogram. Check your answer by clicking the `Grade` button\n",
+    "\n",
+    "<!-- ::: q-block.reminder -->\n",
+    "\n",
+    "### Hints\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 1</summary>\n",
+    "    The NOT, CNOT, and Hadamard gates are their own inverse. That means that if two of these gates are placed side by side they can simply be taken off.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 2</summary>\n",
+    "    The SWAP gate can be taken off if the subsequent operations are adjusted between the two qubits.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 3</summary>\n",
+    "    If a CNOT has the same control and target as another CNOT for which two NOT gates are applied before and after the control qubit, this can be simplified to a single NOT gate on the target qubit of the CNOT as a NOT gate is applied to the target whether the control qubit is initially in state 0 or 1.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 4</summary>\n",
+    "    The circuit can be simplified until only three gates remain in the algorithm.\n",
+    "</details>\n",
+    "<!-- ::: -->\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "goals": [
+     {
+      "id": "enigma-quiz-1"
+     }
+    ],
+    "grader_answer": "qc",
+    "grader_id": "quantum_enigmas/problem1"
+   },
+   "outputs": [],
+   "source": [
+    "from qiskit import QuantumCircuit, Aer, transpile\n",
+    "from qiskit.visualization import plot_histogram\n",
+    "\n",
+    "# Start your work here.\n",
+    "# We've provided the circuit that is shown above\n",
+    "# Your circuit MUST be named qc\n",
+    "# Remove all barriers before submitting for grading\n",
+    "\n",
+    "qc = QuantumCircuit(3)\n",
+    "\n",
+    "qc.h(0)\n",
+    "qc.h(2)\n",
+    "qc.cx(0, 1)\n",
+    "qc.barrier(0, 1, 2)\n",
+    "qc.cx(2, 1)\n",
+    "qc.x(2)\n",
+    "qc.cx(2, 0)\n",
+    "qc.x(2)\n",
+    "qc.barrier(0, 1, 2)\n",
+    "qc.swap(0, 1)\n",
+    "qc.x(1)\n",
+    "qc.cx(2, 1)\n",
+    "qc.x(0)\n",
+    "qc.x(2)\n",
+    "qc.cx(2, 0)\n",
+    "qc.x(2)\n",
+    "\n",
+    "# Execute the circuit and draw the histogram\n",
+    "\n",
+    "measured_qc = qc.measure_all(inplace=False)\n",
+    "backend = Aer.get_backend('qasm_simulator') # the device to run on\n",
+    "result = backend.run(transpile(measured_qc, backend), shots=1024).result()\n",
+    "counts  = result.get_counts(measured_qc)\n",
+    "plot_histogram(counts)"
```

Comment:
This should also be an acceptable answer when the GRADE button is pressed

qc.h(1)
qc.h(2)
qc.cx(1,0)